Floofbot/Modules/Fun.cs

```diff
@@ -168,5 +168,172 @@ public async Task RequestCatFact()
             }
             await Context.Channel.SendMessageAsync(fact);
         }
+
+        [Command("minesweeper")]
+        [Summary("Minesweeper minigame")]
+        public async Task Minesweeper([Summary("grid x")]int gridx, [Summary("grid y")]int gridy, [Summary("bomb count")]int bombs)
+        {
+           
+            //no negative numbers!
+            if(gridx < 1 || gridy < 1 || bombs < 0)
+            {
+                await Context.Channel.SendMessageAsync("Invalid grid size or bomb count");
+                return;
+            }
+            
+            //limits the size of the board
+            if(gridx > 10 || gridy > 10)
+            {
+                await Context.Channel.SendMessageAsync("Max Grid Size: 10 x 10");
+                return;
+            }
+            else if(bombs >= gridy * gridx)
+            {
+                await Context.Channel.SendMessageAsync("Too many bombs!");
+                return;
+            }
+
+            Gameboard game = new Gameboard(gridx, gridy, bombs);
+            EmbedBuilder builder = new EmbedBuilder();
+            builder.Title = ":bomb: Minesweeper";
+            builder.Color = EMBED_COLOR;
+            builder.Description = game.getBoard();
+
+            await Context.Channel.SendMessageAsync("", false, builder.Build());
+        }
+
+        public class Gameboard
```

Comment:
I don't think it needs to be a public class

Reply:
lol I'd actually prefer if this were its own class in its own file.
Also, I'd prefer the capitalization GameBoard*
I don't think we currently have a place to put this class, but maybe Floofbot/Activities/GameBoard.cs works.
Then, each individual activity could become a class, since Fun.cs will soon become a god object.

Reply:
I think I would prefer /Modules/Helpers/GameBoard.cs if we do this approach

Reply:
I needs to be public otherwise it will error out with the class being inaccessible